source/module_elecstate/module_dm/test/test_dm_R_init.cpp

```diff
@@ -276,6 +277,38 @@ TEST_F(DMTest, DMInit4)
     delete kv;
 }
 
+// test for save_DMR
+TEST_F(DMTest, saveDMR)
+{
+    // initalize a kvectors
+    K_Vectors* kv = nullptr;
+    int nspin = 2;
+    int nks = 4; // since nspin = 2
+    kv = new K_Vectors;
```

Comment:
Maybe you can directly create an instance of `K_Vectors` class instead of allocate memory in heap explicitly. If you really want some features of pointer, I would suggest use `std::shared_ptr<K_Vectors>` or `std::unique_ptr<K_Vectors>`, they are C++ 11 smart pointer and can manage memory for you automatically, which means, more safe, but present is also okay because this test code is simple

Reply:
Forgot to delete `kv`, I have added it.

Reply:
> Maybe you can directly create an instance of `K_Vectors` class instead of allocate memory in heap explicitly. If you really want some features of pointer, I would suggest use `std::shared_ptr<K_Vectors>` or `std::unique_ptr<K_Vectors>`, they are C++ 11 smart pointer and can manage memory for you automatically, which means, more safe, but present is also okay because this test code is simple

Forgot to delete kv, I have added it.